### PR TITLE
feat(component): `<InlineVideo />`

### PR DESCRIPTION
**Pull request by EnMod**
🎟️ [Asana Task](https://app.asana.com/0/0/1203260018123853/f)
🔍 [Preview Link](https://react-components-git-nq-inlinevideo-hashicorp.vercel.app/components/inlinevideo)

---

## Description

This PR creates an `<InlineVideo />` component, and bumps `mktg-global-styles` to `4.3.1` so the component can properly render background gradients. Will be used soon in #784 .

### Tech debt

Testing for this PR and #784 will be handled in a separate PR after both are merged to the assembly for these components. 

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/inline-video/style.module.css [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    --gradient-stops: #f26c64 0%, #ffe68d 100%;
+  }
+
+  &.networking {
+    --gradient-stops: #c750b7 0%, #ffb3b7 100%;
+  }
+
+  &.applications {
+    --gradient-stops: #5493e4 0%, #5bd0d9 50%, #a3f7d7 100%;
+  }
```
Comment (alexcarpenter) [inline packages/inline-video/style.module.css:54]: these are available via wpl tokens https://github.com/hashicorp/mktg-global-styles/blob/main/custom-properties/color.css#L132-L191

File: packages/inline-video/style.module.css [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  @media (--medium-up) {
+    @nest .left & {
+      padding-left: var(--gradient-size);
+    }
+
+    @nest .right & {
+      padding-right: var(--gradient-size);
+    }
+
+    &::before {
```
Comment (alexcarpenter) [inline packages/inline-video/style.module.css:20]: Thoughts on just making this gradient the full width of the element and changing the padding left/right based on placement?

Comment (EnMod) [reply packages/inline-video/style.module.css:20]: Ah, I think I know what you mean! I'll give it a go and see how that looks

Comment (EnMod) [reply packages/inline-video/style.module.css:20]: It ended up working well! The changes are captured in the latest commit (along with a fix for a regression 😅)

File: packages/inline-video/style.module.css (hunk 1 above)
Comment (EnMod) [reply packages/inline-video/style.module.css]: @alexcarpenter Hmm...I've replaced all the values here with their proper tokens, and yet the actual declaration doesn't apply the value initially assigned to `--gradient`? Changing the `solution` prop to a supported value _does_ allow the correct gradient to show, however. 🤔 

Maybe I'm running into some kind of scoping or syntax issue?

Comment (EnMod) [reply packages/inline-video/style.module.css]: Update: Just needed to update deps! We're good to go as of the latest commit:
https://github.com/hashicorp/react-components/pull/789/commits/65a6e3c9295c7a3fed292c039321bc5ae3469648

File: packages/inline-video/types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface InlineVideoProps {
+  appearance?: 'light' | 'dark'
+  url: string
+  description?: string
+  solution?: 'infrastructure' | 'security' | 'networking' | 'applications'
+  gradientSide?: 'left' | 'right'
```
Comment (alexcarpenter) [inline packages/inline-video/types.ts:6]: 💭 How about `gradientPosition` vs `gradientSide`?

File: packages/inline-video/index.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  return (
+    <div
+      className={classNames(s.inlineVideo, s[appearance], s[gradientSide])}
+      data-testid="wpl-inline-video"
+    >
+      <div className={classNames(s.videoContainer, solution && s[solution])}>
+        <div className={s.video}>
+          {url.includes('wistia') ? (
+            <WistiaPlayer {...playerProps} />
```
Comment (alexcarpenter) [inline packages/inline-video/index.tsx:30]: On wistia videos, I think we want to hide controls on load until video is played. Ref: https://github.com/hashicorp/hashicorp-www-next/blob/main/pages/about/index.tsx#L198-L202

```js
config={{
  options: {
    controlsVisibleOnLoad: false,
  },
}}
```

Comment (alexcarpenter) [reply packages/inline-video/index.tsx:30]: CleanShot 2022-11-01 at 13 16 34@2x

Comment (EnMod) [reply packages/inline-video/index.tsx]: Good call, makes sense to keep that behavior consistent :+1: 

I've addressed this here:
https://github.com/hashicorp/react-components/pull/789/commits/4decdef0a19f180150dfe468ff095216e2c9e5e3

File: packages/inline-video/types.ts (hunk 3 above)
Comment (EnMod) [reply packages/inline-video/types.ts]: I like it! Addressed here:
https://github.com/hashicorp/react-components/pull/789/commits/3ad996e88610cc6d7b68231f208620674a254ba1

File: packages/inline-video/index.tsx (hunk 4 above)
Comment (EnMod) [reply packages/inline-video/index.tsx]: Good point! I've addressed that here:
https://github.com/hashicorp/react-components/pull/789/commits/4decdef0a19f180150dfe468ff095216e2c9e5e3

File: packages/inline-video/index.test.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import InlineVideo from '.'
+
+const defaultProps = {
+  url: 'https://youtu.be/nJ0aI6sEDRo',
+  description:
+    'Description - 2 lines with character limit of 90. Tempus in egestas sagittis nulla feugiat',
+}
+
+describe('<InlineVideo />', () => {
+  it('TODO: Write tests', () => {
```
Comment (alexcarpenter) [inline packages/inline-video/index.test.js:11]:
```suggestion
  it('renders the provided description text', () => {
```

**Review by alexcarpenter — APPROVED** [review]
LGTM, I think we can just update the test name to reflect checking the description presence and we're good for an initial release.

**Pull request merged**